### PR TITLE
feat: automatic next button label updation

### PR DESCRIPTION
**Pull request by Dhruwang**
## What does this PR do?

Fixes [1490](https://linear.app/formbricks/issue/FOR-1490/ux-tweak-back-button-label-get-set-for-all-questions-next-button-label)

Screenshot 2024-01-02 at 9 40 18 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

Change next button label for a question
It will automatically update all next button label for all question where next button label value is blank
You can also change it for particular questions
### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://formbricks.com/clmyhzfrymr4ko00hycsg1tvx) Without it we wont be able to merge it 🙏

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx [hunk 1]
```diff
@@ -98,6 +98,23 @@ export default function QuestionCard({
...
+      if (!q.buttonLabel || q.buttonLabel?.trim() === "" || questionsWithEmptyLabel.includes(q.id)) {
+        if (!questionsWithEmptyLabel.includes(q.id)) {
+          setquestionsWithEmptyLabel((prevArray) => [...prevArray, q.id]);
+        }
+        updateQuestion(index, { buttonLabel: labelValue });
+      } else {
+        updateQuestion(questionIdx, { buttonLabel: labelValue });
+      }
+    });
+  };
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx:114]: please make sure to add an empty line between two functions to improve readability

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx [hunk 2]
```diff
@@ -307,8 +324,7 @@ export default function QuestionCard({
                               maxLength={48}
                               placeholder={lastQuestion ? "Finish" : "Next"}
                               onChange={(e) => {
-                                if (e.target.value.trim() == "") e.target.value = "";
-                                updateQuestion(questionIdx, { buttonLabel: e.target.value });
+                                editNextButtonLabel(e.target.value);
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx:327]: I'm not really satisfied that we loop through all questions `onChange` (with every character change). Can we also do this e.g. `onBlur`?

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx [hunk 3]
```diff
@@ -98,6 +98,23 @@ export default function QuestionCard({
   const question = localSurvey.questions[questionIdx];
   const open = activeQuestionId === question.id;
   const [openAdvanced, setOpenAdvanced] = useState(question.logic && question.logic.length > 0);
+  const [questionsWithEmptyLabel, setquestionsWithEmptyLabel] = useState<string[]>([]);
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx:101]: please fix camel case to `setQuestionWithEmptyLabel`

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx [hunk 4]
```diff
@@ -98,6 +98,23 @@ export default function QuestionCard({
...
+          setquestionsWithEmptyLabel((prevArray) => [...prevArray, q.id]);
+        }
+        updateQuestion(index, { buttonLabel: labelValue });
+      } else {
+        updateQuestion(questionIdx, { buttonLabel: labelValue });
+      }
+    });
+  };
+  useEffect(() => {
+    setquestionsWithEmptyLabel([]);
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx:116]: I'm not getting why we need the `questionsWithEmptyLabel` state. Why isn't it working without it?

**Review by mattinannt — CHANGES_REQUESTED** [review]
@Dhruwang thanks for the fix and also for providing a german example in the screenshot 😅💪
There are a few things I would like to discuss or see changed before we can merge it 😊

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx (hunk 4 above)
Comment (Dhruwang) [reply apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx:116]: Actually this state was used to update next button value of a question, which has been already set
But this was required on with if we check for `emptyNextButtonLabel` with `onChange`
`onBlur` makes it much simpler 😊💪

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx (hunk 2 above)
Comment (Dhruwang) [reply apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx:327]: That works 😊

**Review by mattinannt — APPROVED** [review]
@Dhruwang thank you for the fixes; looks great! 😊💪

**Pull request merged**